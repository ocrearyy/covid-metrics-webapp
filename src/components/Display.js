import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import thunkFunction from '../redux/API';

const Display = () => {
  const country = useSelector((store) => store);
  useEffect(() => {
    thunkFunction();
  }, []);
  const details = window.location.href.slice(30);
  return (
    <div className="display-body">
      <div>
        <h1 className="main-text">Further Details</h1>
      </div>
      <section className="section">
        {country.map((i) => (i.id === details
          ? (
            <div className="main-p" key={i.id}>
              <h1 key={i.id}>{i.id.toUpperCase()}</h1>
              <p>
                Today&lsquo;s Confirmed Cases:
                &nbsp;
                {i.today_confirmed}
              </p>
              <p>
                Today&lsquo;s Confirmed deaths:
                &nbsp;
                {i.today_deaths}
              </p>
              <p>
                Today&lsquo;s Open Cases:
                &nbsp;
                {i.today_open_cases}
              </p>
              <p>
                Today&lsquo;s Recovered Cases:
                &nbsp;
                {i.today_recovered}
              </p>
            </div>
          )
           :console.log('it is not right match')))}

      </section>
    </div>
  );
};

export default Display;
