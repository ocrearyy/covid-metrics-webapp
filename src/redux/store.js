import { configureStore } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import { logger } from 'redux-logger';
import continentReducer from './continent';

export default configureStore({
  reducer: continentReducer,
  middleware: [thunk, logger],
});
