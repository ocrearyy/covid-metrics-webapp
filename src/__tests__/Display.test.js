import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Display from '../components/Display';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Display />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
