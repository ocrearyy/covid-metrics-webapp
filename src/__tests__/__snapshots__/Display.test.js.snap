// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="display-body"
>
  <div>
    <h1
      className="main-text"
    >
      Further Details
    </h1>
  </div>
  <section
    className="section"
  />
</div>
`;
